components/TableOfContents.tsx

```
@@ -53,27 +144,58 @@ const TableOfContents: React.FC<TableOfContentsProps> = ({ items }) => {
+      {/* Top scroll indicator */}
+      {canScrollUp && (
+        <div className="absolute top-0 left-0 right-0 h-4 bg-gradient-to-b from-white dark:from-secondary-800 to-transparent pointer-events-none z-10 flex items-start justify-center">
+          <ChevronUpIcon className="w-4 h-4 text-secondary-400 dark:text-secondary-500 mt-0.5 scroll-indicator" />
+        </div>
+      )}
+
+      {/* Scrollable content */}
+      <div
+        ref={scrollContainerRef}
+        className={`toc-scrollable ${heightClass} overflow-y-auto`}
+        tabIndex={0}
```

Biome (1.9.4)

[error] The HTML element div is non-interactive. Do not use tabIndex.

Adding non-interactive elements to the keyboard navigation flow can confuse users.
Unsafe fix: Remove the tabIndex attribute.


(lint/a11y/noNoninteractiveTabindex)